src/client/dfuse/pil4dfs/int_dfs.c

```diff
@@ -674,90 +672,97 @@ retrieve_handles_from_fuse(int idx)
 
 	fd = libc_open(dfs_list[idx].fs_root, O_RDONLY | O_DIRECTORY | O_NOFOLLOW);
 	if (fd < 0) {
-		if (bLog)
-			D_ERROR("failed to open dir: %s", dfs_list[idx].fs_root);
+		errno_saved = errno;
+		D_ERROR("failed to open dir %s: %s\n", dfs_list[idx].fs_root, strerror(errno));
+		errno = errno_saved;
 		return (-1);
 	}
 
 	cmd = _IOR(DFUSE_IOCTL_TYPE, DFUSE_IOCTL_REPLY_SIZE, struct dfuse_hs_reply);
 	rc  = ioctl(fd, cmd, &hs_reply);
 	if (rc != 0) {
-		if (bLog)
-			D_ERROR("failed to query size info from dfuse with ioctl().");
+		errno_saved = errno;
+		D_ERROR("failed to query size info from dfuse with ioctl(): %s\n", strerror(errno));
 		goto err;
 	}
 
 	/* To determine the size of buffer we need to accommodate the data from fuse */
 	buff_size = max(hs_reply.fsr_pool_size, hs_reply.fsr_cont_size);
 	buff_size = max(buff_size, hs_reply.fsr_dfs_size);
 	D_ALLOC(buff, buff_size);
-	if (buff == NULL)
+	if (buff == NULL) {
+		errno_saved = ENOMEM;
+		D_ERROR("failed to allocate memory.\n");
```

Comment:
D_ALLOC() will log itself on failure, you don't need to log again.

Reply:
Fixed as suggested. Thank you!